drivers/storage/portworx/component/plugin.go

```diff
@@ -0,0 +1,359 @@
+			return true
+		}
+	}
+
+	return false
+}
+
+func (p *plugin) Reconcile(cluster *corev1.StorageCluster) error {
+
+	ownRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	var errList []string
+	//create nginx resources
+	if err := p.createConfigmap(nginxConfigMapFileName, cluster.Namespace, ownRef); err != nil {
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating nginx configmap %s ", err)
+	}
+	if err := p.createDeployment(nginxDeploymentFileName, cluster.Namespace, ownRef); err != nil {
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating nginx configmap %s ", err)
+	}
+	if err := p.createService(nginxServiceFileName, cluster.Namespace, ownRef); err != nil {
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating nginx configmap %s ", err)
+	}
+
+	//create portworx plugin resources
+	if err := p.createDeployment(pluginDeploymentFileName, cluster.Namespace, ownRef); err != nil {
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating deployment %s ", err)
+	}
+	if err := p.createService(pluginServiceFileName, cluster.Namespace, ownRef); err != nil {
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating service %s ", err)
+	}
+	if err := p.createConfigmap(pluginConfigmapFileName, cluster.Namespace, ownRef); err != nil {
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating config map  %s ", err)
+	}
+
+	//create console plugin
+	if err := p.createConsolePlugin(consolePluginFileName, cluster.Namespace, ownRef); err != nil {
+		errList = append(errList, err.Error())
+		logrus.Errorf("error during creating console plugin %s ", err)
+	}
+
+	if len(errList) > 0 {
+		return commonerrors.New(strings.Join(errList, " , "))
+	}
+	return nil
+}
+
+func (p *plugin) Delete(cluster *corev1.StorageCluster) error {
+	var errList []string
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	//delete plugin configs
+	if err := k8s.DeleteDeployment(p.client, PluginDeloymentName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteConfigMap(p.client, PluginConfigMapName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteService(p.client, PluginServiceName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+
+	//delete nginx configs
+	if err := k8s.DeleteDeployment(p.client, NginxDeploymentName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteConfigMap(p.client, NginxConfigMapName, cluster.Namespace, *ownerRef); err != nil {
+		errList = append(errList, err.Error())
+	}
+	if err := k8s.DeleteService(p.client, NginxServiceName, cluster.Namespace, *ownerRef); err != nil {
```

Comment:
Can you verify if the certificate secret that gets created with both the services also gets deleted when the service is deleted?